packages/grafana-ui/src/components/Combobox/useComboboxFloat.ts

```diff
@@ -59,17 +60,30 @@ export const useComboboxFloat = (items: Array<ComboboxOption<string | number>>,
   });
 
   const longestItemWidth = useMemo(() => {
+    const maxItemsLength = Math.min(items.length, WIDTH_CALCULATION_LIMIT_ITEMS);
+    const itemsToLookAt = items.slice(0, maxItemsLength);
+
     let longestItem = '';
-    const itemsToLookAt = Math.min(items.length, WIDTH_CALCULATION_LIMIT_ITEMS);
+    let withIcon = false;
+    let withDescription = false;
 
-    for (let i = 0; i < itemsToLookAt; i++) {
-      const itemLabel = items[i].label ?? items[i].value.toString();
+    for (const item of itemsToLookAt) {
+      const itemLabel = item.label ?? item.value.toString();
       longestItem = itemLabel.length > longestItem.length ? itemLabel : longestItem;
+      withIcon ||= !!item.imgUrl;
+      withDescription ||= !!item.description;
     }
 
-    const size = measureText(longestItem, MENU_ITEM_FONT_SIZE, MENU_ITEM_FONT_WEIGHT).width;
+    const textWidth = measureText(longestItem, MENU_ITEM_FONT_SIZE, MENU_ITEM_FONT_WEIGHT).width;
+
+    let adjustedSize = textWidth + MENU_ITEM_PADDING * 2 + scrollbarWidth;
+    if (withIcon && withDescription) {
+      adjustedSize += MENU_OPTION_HEIGHT_DESCRIPTION - MENU_ITEM_PADDING;
+    } else if (withIcon) {
+      adjustedSize += MENU_OPTION_HEIGHT - MENU_ITEM_PADDING;
+    }
```

Comment:
I'm a bit confused, but I don't 100% remember what all this does. Maybe @tskarhed can help refresh my memory?

I thought this was trying to find the _longest_ item to size the _width_ of the menu dropdown. I don't think this needs to use the `_HEIGHT` variables?

Reply:
You're right, they are only used in the height estimation.